### PR TITLE
make a lib for frost-coordinator, and move Coordinator and related types into it

### PR DESCRIPTION
**Pull request by xoloki**

### CODE REVIEW

File: frost-coordinator/src/coordinator.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
+use std::time::Duration;
```
Comment (donpdonp) [inline frost-coordinator/src/coordinator.rs:1]: nitpick: it'd be nice to have this done with git mv main.rs coordinator.rs, then edit coordinator, to preserve more of the edit history/ownership.

Comment (donpdonp) [reply frost-coordinator/src/coordinator.rs:1]: then main.rs can be a new creation.

Comment (xoloki) [reply frost-coordinator/src/coordinator.rs:1]: I need to split main.rs into coordinator.rs and main.rs, so git mv doesn't seem right...

Comment (xoloki) [reply frost-coordinator/src/coordinator.rs:1]: Seems like I'm gonna lose history in one or the other

Comment (donpdonp) [reply frost-coordinator/src/coordinator.rs:1]: agreed, though its 10 lines of minor bits like use and const statements in main vs all of coordinator.rs. its a nitpick so either way is okay with me.

File: frost-coordinator/src/coordinator.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    #[command(subcommand)]
+    pub command: Command,
+}
+
+#[derive(clap::Subcommand, Debug)]
+pub enum Command {
+    Dkg,
+    Sign { msg: String },
+    GetAggregatePublicKey,
+}
```
Comment (netrome) [inline frost-coordinator/src/coordinator.rs:22]: Nit: Imo, the Cli type belongs to `main` and not the library. I also think the `.run()` function could be moved back to `main.rs`.

**Review by netrome — APPROVED** [review]
LGTM, just a minor optional nit

File: frost-coordinator/src/coordinator.rs (hunk 2 above)
Comment (xoloki) [reply frost-coordinator/src/coordinator.rs]: Cli moved

Comment (xoloki) [reply frost-coordinator/src/coordinator.rs]: Not sure if `run()` belongs in main, it's just dispatching to other Coordinator `fn`

**Review by donpdonp — APPROVED** [review]

**Pull request merged**